pvlib/pvsystem.py

```diff
@@ -1475,6 +1448,97 @@ def get_iam(self, aoi, iam_model='physical'):
         else:
             raise ValueError(model + ' is not a valid IAM model')
 
+    def get_cell_temperature(self, poa_global, temp_air, wind_speed, model,
+                             **kwargs):
+        """
+        Determine cell temperature using the method specified by ``model``.
+
+        Parameters
+        ----------
+        poa_global : numeric
+            Total incident irradiance [W/m^2]
+
+        temp_air : numeric
+            Ambient dry bulb temperature [C]
+
+        wind_speed : numeric
+            Wind speed [m/s]
+
+        model : str
+            Supported models include ``'sapm'``, ``'pvsyst'``,
+            ``'faiman'``, ``'fuentes'``, and ``'noct_sam'``
+
+        **kwargs
+            Extra arguments passed to the model function.
+
+        Returns
+        -------
+        numeric
+            values in degrees C.
+
+        Notes
+        -----
+        Some temperature models have requirements for the input types;
+        see the documentation of the underlying model function for details.
+        """
+        def _build_args(keys):
```

Comment:
maybe put this in `pvlib.tools`? OK if that is deferred until another use arises.